app/api/revalidate/route.ts

```diff
@@ -0,0 +1,91 @@
+import crypto from 'crypto'
+import { fetchContentPage } from '@lib/strapi/contentpage'
+import { fetchSection } from '@lib/strapi/contentSection'
+import { PageType } from '@models/page'
+import { SingleResponse } from '@lib/strapi'
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+import { revalidatePath, revalidateTag } from 'next/cache'
+
+const API_KEY = process.env.API_KEY
+
+export async function POST(request: NextRequest) {
+  const bearer = request.headers
+    .get('authorization')
+    ?.split('Bearer:')[1]
+    ?.trim()
+  if (
+    bearer &&
+    API_KEY &&
+    crypto.timingSafeEqual(Buffer.from(bearer), Buffer.from(API_KEY))
+  ) {
+    return NextResponse.json({ error: 'Invalid token' }, { status: 401 })
+  }
+
+  try {
+    const tag = request.nextUrl.searchParams.get('tag')
+    if (tag) {
+      // Manual API call
+      revalidateTag(tag)
```

Comment:
For revalidateTag to work the tags have to be provided in the fetch request, so currently this doesn't do anything.

Reply:
Yeah, my thought was that this will be properly done in #45